infra/cifuzz/actions/entrypoint.py

```diff
@@ -49,33 +50,44 @@ def main():
   pr_ref = os.environ.get('GITHUB_REF')
   commit_sha = os.environ.get('GITHUB_SHA')
   event = os.environ.get('GITHUB_EVENT_NAME')
-
-  # Get the shared volume directory and create required directorys.
   workspace = os.environ.get('GITHUB_WORKSPACE')
+
+  # Check if failures should be reported.
+  failure_allowed = (os.environ.get('FAILURE_ALLOWED').lower() == 'true')
+
+  error_code = 1
+  if not failure_allowed:
+    out_dir = os.path.join(workspace, 'out')
+    os.makedirs(out_dir, exist_ok=True)
+    file_handle = open(os.path.join(out_dir, 'testcase'), 'a')
+    file_handle.write('No bugs detected.')
+    file_handle.close()
+    error_code = 0
+
   if not workspace:
     logging.error('This script needs to be run in the Github action context.')
-    return 1
+    return error_code
 
   if event == 'push' and not cifuzz.build_fuzzers(
       oss_fuzz_project_name, github_repo_name, workspace,
       commit_sha=commit_sha):
     logging.error('Error building fuzzers for project %s with commit %s.',
                   oss_fuzz_project_name, commit_sha)
-    return 1
+    return error_code
   if event == 'pull_request' and not cifuzz.build_fuzzers(
       oss_fuzz_project_name, github_repo_name, workspace, pr_ref=pr_ref):
     logging.error('Error building fuzzers for project %s with pull request %s.',
                   oss_fuzz_project_name, pr_ref)
-    return 1
+    return error_code
 
   # Run the specified project's fuzzers from the build.
   run_status, bug_found = cifuzz.run_fuzzers(oss_fuzz_project_name,
                                              fuzz_seconds, workspace)
   if not run_status:
     logging.error('Error occured while running fuzzers for project %s.',
                   oss_fuzz_project_name)
-    return 1
-  if bug_found:
+    return error_code
+  if bug_found and failure_allowed:
```

Comment:
what if `bug_found and not failure_allowed`? can we still log the message but `return 0` and not mark the action as failed?

Reply:
Fixed.